### ISSUE TITLE
Make a windows compatible "make.sh" (make.bat ?)

### ISSUE THREAD

**Issue by riccardobl** [jmePhonon/jmePhonon#9]

**Comment by MeFisto94** [comment]
I don't know how viable this approach is since docker for windows is relatively new and might come with some caveats.
I wonder why it's even necessary considering the build running gradle (which however relies on prepare.sh which would also need to be ported or implemented in gradle, which looks absolutely possible since it only downloads and unzips stuff mostly).

But since it also runs gcc it's probably the best to use a linux system after all :D 
Sorry if this comment makes no sense, it's just my thoughts so far

**Comment by riccardobl** [comment]
Yeah, most of the stuff done in prepare.sh can be done in gradle aswell (even thought it's easier to do it in bash) but even then you would have to install mingw to compile the natives. 

The docker image is more handy, since it has the proper versions of everything needed to compile for every supported os.

But I'm not sure how hard it is to set docker on windows, it will have to run on a virtual machine and be able to mount the project folder from the host.

There is some doc here https://docs.docker.com/docker-for-windows/install/ but i didn't have a chance to try it yet.

**Comment by riccardobl** [comment]
The master now has a make.bat that can build for windows with git bash and docker.
Conveniently Docker toolbox comes bundled together with git bash, so that's the only thing needed to build the project.

The story is different for windows 10 pro or enterprise that can run the newer Docker for Windows that uses Hyper-V, more steps are needed in order to get this to work, git bash has to be installed separately and docker must be configured to share the C driver, this doesn't seem to be always possible for every installation (investigation needed).

There might be a better/cleaner approach to run test compilations for windows only, like running gradle tasks directly from mingw msis.
